File: user/src/main/scala/ru/tdd/app/utils/DateUtils.scala
package ru.tdd.app.utils

import java.time.{LocalDate, LocalDateTime}
import java.util.Date

/**
 * @author Tribushko Danil
 * @since 19.11.2025
 *        Утилиты для работы со временем
 */
trait DateUtils {

  val Second: Int = 1000

  val Minute: Int = Second * 60

  val Hour: Int = Minute * 60

  val Day: Int = Hour * 24

  val NOW: Date = new Date()

  val NOW_TIME: LocalDateTime = LocalDateTime.now()

  val NOW_DATE: LocalDate = LocalDate.now()

  implicit class DateOps(date: Date) {

    def +(time: Long) = new Date(date.getTime + time)

    def <(value: Date): Boolean = date.before(value)

    def >(value: Date): Boolean = date.after(value)
  }
}

File: user/src/main/java/ru/tdd/user/application/utils/DateUtils.java
package ru.tdd.user.application.utils;

import java.util.Date;

/**
 * @author Tribushko Danil
 * @since 01.02.2026
 * Набор утилит для работы с датой
 */
public class DateUtils {

    public static final long SECUND = 1000;

    public static final long MINUTE = 60 * SECUND;

    public static final long HOUR = 60 * MINUTE;

    public static final long DAY = 24 * HOUR;

    private DateUtils(){}

    public static Date plusTime(Date date1, long time) {
        return new Date(date1.getTime() + time);
    }

    public static boolean isFuture(Date date) {
        return date.after(new Date());
    }
}
